cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,351 @@
+		return err
+	}
+	t.sinkNode = actorSinkNode
+	if t.cyclicEnabled {
+		c = func() *pipeline.Message {
+			return cCtx.tryGetProcessedMessage()
+		}
+	} else {
+		c = func() *pipeline.Message {
+			return sCtx.tryGetProcessedMessage()
+		}
+	}
+	d = func(ctx context.Context, msg pipeline.Message) (bool, error) {
+		return actorSinkNode.HandleMessage(sCtx, msg)
+	}
+	t.nodes = append(t.nodes, NewActorNode(c, d))
+
+	t.started = true
+	log.Info("table actor is started", zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID))
+	return nil
+}
+
+func (t *tableActor) stop(err error) {
+	if t.stopped {
+		return
+	}
+	t.stopped = true
+	t.err = err
+	t.cancel()
+	log.Info("table actor will be stopped",
+		zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Error(err))
+}
+
+func (t *tableActor) checkError() {
+	if t.err != nil {
+		t.reportErr(t.err)
+		t.err = nil
+	}
+}
+
+// ============ Implement TablePipline, must be threadsafe ============
+
+// ResolvedTs returns the resolved ts in this table pipeline
+func (t *tableActor) ResolvedTs() model.Ts {
+	// TODO: after TiCDC introduces p2p based resolved ts mechanism, TiCDC nodes
+	// will be able to cooperate replication status directly. Then we will add
+	// another replication barrier for consistent replication instead of reusing
+	// the global resolved-ts.
+	if redo.IsConsistentEnabled(t.replConfig.Consistent.Level) {
+		return t.sinkNode.ResolvedTs()
+	}
+	return t.sortNode.ResolvedTs()
+}
+
+// CheckpointTs returns the checkpoint ts in this table pipeline
+func (t *tableActor) CheckpointTs() model.Ts {
+	return t.sinkNode.CheckpointTs()
+}
+
+// UpdateBarrierTs updates the barrier ts in this table pipeline
+func (t *tableActor) UpdateBarrierTs(ts model.Ts) {
+	msg := message.BarrierMessage(ts)
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	if err != nil {
+		log.Warn("send fails", zap.Reflect("msg", msg), zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Error(err))
+	}
+}
+
+// AsyncStop tells the pipeline to stop, and returns true is the pipeline is already stopped.
+func (t *tableActor) AsyncStop(targetTs model.Ts) bool {
+	msg := message.StopSinkMessage()
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	log.Info("send async stop signal to table", zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Uint64("targetTs", targetTs))
+	if err != nil {
+		if cerror.ErrMailboxFull.Equal(err) {
+			return false
+		}
+		if cerror.ErrSendToClosedPipeline.Equal(err) {
+			return true
+		}
+		log.Panic("send fails", zap.Reflect("msg", msg), zap.Error(err))
+	}
+	return true
+}
+
+// Workload returns the workload of this table
+func (t *tableActor) Workload() model.WorkloadInfo {
+	// We temporarily set the value to constant 1
+	return workload
+}
+
+// Status returns the status of this table pipeline
```

Comment:
// Status returns the status of this table actor

Reply:
actor implemented the table pipeline interface, it's better to say table pipeline here